specification/relay/resource-manager/Microsoft.Relay/stable/2021-11-01/Namespaces.json

```diff
@@ -0,0 +1,982 @@
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/PrivateEndpointConnectionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Private Endpoint Connection successfully deleted."
+          },
+          "202": {
+            "description": "Private Endpoint Connection  delete request accepted."
+          },
+          "204": {
+            "description": "No content."
+          },
+          "default": {
+            "description": "Relay error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../common/v2/definitions.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "get": {
+        "tags": [
+          "Namespaces PrivateEndpointConnections"
+        ],
+        "operationId": "PrivateEndpointConnections_Get",
+        "x-ms-examples": {
+          "NameSpacePrivateEndPointConnectionGet": {
+            "$ref": "./examples/PrivateEndpointConnections/PrivateEndpointConnectionsGet.json"
+          }
+        },
+        "description": "Gets a description for the specified Private Endpoint Connection name.",
+        "externalDocs": {
+          "url": "https://msdn.microsoft.com/en-us/library/azure/mt639379.aspx"
+        },
+        "parameters": [
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/NamespaceNameParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/PrivateEndpointConnectionNameParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Private Endpoint Connection  successfully returned.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnection"
+            }
+          },
+          "default": {
+            "description": "Relay error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../common/v2/definitions.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Relay/namespaces/{namespaceName}/privateLinkResources": {
```

Comment:
>"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Relay/namespaces/{namespaceName}/privateLinkResources" [](http://example.com/codeflow?start=4&length=142)

how are privateLinkResources created? You are modeling a collection GET without a single GET, you need to model a singleton, see here for reference:

[https://github.com/Azure/azure-rest-api-specs/blob/fd233b0ce344369e14c6a5d16806eb4fe9b77cf4/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/deploymentScripts.json#L274](https://github.com/Azure/azure-rest-api-specs/blob/fd233b0ce344369e14c6a5d16806eb4fe9b77cf4/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/deploymentScripts.json#L274)

[https://github.com/Azure/azure-rest-api-specs/blob/fd233b0ce344369e14c6a5d16806eb4fe9b77cf4/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/deploymentScripts.json#L316](https://github.com/Azure/azure-rest-api-specs/blob/fd233b0ce344369e14c6a5d16806eb4fe9b77cf4/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/deploymentScripts.json#L316)

Reply:
See [Inconsistent or incomplete CRUD](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review) for more reference on GET for collection and single resource.

Reply:
hi @jorgecotillo,
this API was required for to implement the Private endpoints feature and privatelinkresource is not created by user, its a static data we returned. 

Storage RP also has implemented it in similar way : https://github.com/Azure/azure-rest-api-specs/blob/main/specification/storage/resource-manager/Microsoft.Storage/stable/2021-09-01/storage.json#L1441